### PR TITLE
Add CodeQL workflow

### LINKED ISSUE #86: Missing CodeQL analysis
### Description of the issue

The repository is missing detailed CodeQL coverage/output. Add a workflow file to rectify this.

### PR DESCRIPTION
**Pull request by HumzahJavid**
Closes #86 

## Description
This PR will:
 - Add a CodeQL workflow file

### CODE REVIEW

**Review by Pasarus — CHANGES_REQUESTED** [review]
Could we remove most of the comments? largely seems to be an exercise of check this, do that, etc, then delete.

**Review by Pasarus — APPROVED** [review]

**Comment by HumzahJavid** [comment]
A decision was made to remove the harden runner from this file as it is not necessary (does not add much value).

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (Pasarus) [inline .github/workflows/codeql.yml:1]: Again can we remove these comments that no longer hold value?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#
+name: 'CodeQL'
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+  schedule:
+    - cron: '0 0 * * 1'
```
Comment (Pasarus) [inline .github/workflows/codeql.yml:15]:
```suggestion
# A random time on a Tuesday once a week.
    - cron: '34 1 * * 2'
```

It's good practice not to have every cronjob use the same time each week as it leads to spikes in Datacenter use and is generally poor practice, so we avoid these sorts of times for that reason, every hour, every day at midnight etc.

**Review by Pasarus — CHANGES_REQUESTED** [review]
Just a missed comment and a quick change.

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (HumzahJavid) [reply .github/workflows/codeql.yml]: I've made the requested changes.

**Review by Pasarus — APPROVED** [review]

**Pull request merged**